test/feature/middleware_support.feature

```diff
@@ -0,0 +1,60 @@
+Feature: Middleware support
+
+  Scenario Outline: simple middleware support
+    Given a file "routes.coffee" with the content
+      """
+      module.exports = ({ GET }) ->
+        GET '/users', to: 'users#index'
+      """
+    And a file "controllers/users_controller.coffee" with the content
+      """
+      middle = (req, res, next) ->
+        req.custom = 'foo'
+        next()
+
+      class UsersController
+        middleware:
+          index: middle
+
+        index: (req, res) -> res.end req.custom
+      module.exports = UsersController
+      """
+    And an exprestive app using defaults
+    When making a <REQUEST> request to "<URL>"
+    Then the response body should be "<RESPONSE BODY>"
+
+    Examples:
+      | REQUEST | URL     | RESPONSE BODY |
+      | GET     | /users  | foo           |
```

Comment:
Or you could use vim with [godlygeek/tabular](https://github.com/godlygeek/tabular) and these tables auto-align themselves to fit their content perfectly :)

(Props to @kevgo for showing me this awesome vim plugin)